cranelift-codegen/meta-python/base/legalize.py

```diff
@@ -375,39 +375,14 @@ def widen_imm(signed, op):
     )
 
 for int_ty in [types.i8, types.i16]:
-    for op in [ushr_imm, ishl_imm]:
+    for op in [ishl, ishl_imm, ushr, ushr_imm, sshr, sshr_imm]:
         widen.legalize(
             a << op.bind(int_ty)(b, c),
             Rtl(
                 x << uextend.i32(b),
                 z << op.i32(x, c),
                 a << ireduce.bind(int_ty)(z)
             ))
-
-    widen.legalize(
-        a << ishl.bind(int_ty)(b, c),
-        Rtl(
-            x << uextend.i32(b),
-            z << ishl.i32(x, c),
-            a << ireduce.bind(int_ty)(z)
-        ))
-
-    widen.legalize(
-        a << ushr.bind(int_ty)(b, c),
-        Rtl(
-            x << uextend.i32(b),
-            z << ushr.i32(x, c),
-            a << ireduce.bind(int_ty)(z)
-        ))
-
-    widen.legalize(
-        a << sshr.bind(int_ty)(b, c),
-        Rtl(
-            x << sextend.i32(b),
```

Comment:
Note this uses `sextend` here, not `uextend`, so it can't be refactored with the rest. I assume `sshr_imm` will need to use `sextend` too.

Reply:
Oops, fixed.